graphqlapi/src/main/java/io/stargate/graphql/schema/types/scalars/FloatCoercing.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.types.scalars;
+
+import graphql.language.FloatValue;
+import graphql.language.IntValue;
+import graphql.schema.Coercing;
+import graphql.schema.CoercingParseLiteralException;
+import graphql.schema.CoercingParseValueException;
+import graphql.schema.CoercingSerializeException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+class FloatCoercing implements Coercing<Float, Float> {
+
+  static FloatCoercing INSTANCE = new FloatCoercing();
+
+  private FloatCoercing() {}
+
+  @Override
+  public Float serialize(Object dataFetcherResult) throws CoercingSerializeException {
+    return (Float) dataFetcherResult;
+  }
+
+  @Override
+  public Float parseValue(Object input) throws CoercingParseValueException {
+    if (input instanceof Double) {
+      double d = (Double) input;
+      float f = (float) d;
+      if (f != d) {
+        throw new CoercingParseLiteralException("Out of bounds");
+      }
```

Comment:
I don't know if rejecting is a good idea. Most systems silently loose precision (CQL included):

```
cqlsh> CREATE KEYSPACE ks1 WITH replication = {'class': 'SimpleStrategy', 'replication_factor' : 1};
cqlsh> CREATE TABLE ks1.tbl1 (id int primary key, value float);
cqlsh> INSERT INTO ks1.tbl1 (id, value) VALUES (1, 12.666428727762776);
cqlsh> SELECT * FROM ks1.tbl1;

 id | value
----+----------
  1 | 12.66643
```

Reply:
Fair point. I think I prefer rejecting but maybe consistency with CQL is more important. What did the go implementation do?

Reply:
It mapped to `float32` using the golang built-in json parser, it lost precision.

Reply:
OK, to be consistent with CQL I'm implementing the following rules:
- float: silent loss of precision, values that are out of range are converted to +/- Infinity
- integer types (bigint, smallint and tinyint): values that are out of range are rejected.